### PR TITLE
Julia: new-version

### PR DESCRIPTION
**Pull request by baberlevi**
not ready for merge quite yet

It installs, but the MPI variant (which is on by default) does not work for me, I'm not sure yet if this is due to something in my environment or something changed in the 1.0.0 package. I don't think I got the older version to install with mpi successfully either.

Also, I'm open to suggestions on a better way to handle the Pkg import that's required on 1.0.0. My solution feels clunky.

Will do some more testing & post back if/when ready for merge.

### CODE REVIEW

**Comment by jcftang** [comment]
I think the way you handle the pkg import is no clunkier than what it was before

File: var/spack/repos/builtin/packages/julia/package.py [hunk 1]
```diff
@@ -197,20 +203,30 @@ def install(self, spec, prefix):
                 juliarc.write('push!(Libdl.DL_LOAD_PATH, "%s")\n' %
                               spec["hdf5"].prefix.lib)
                 juliarc.write('\n')
-            julia("-e", 'Pkg.add("HDF5"); using HDF5')
-            julia("-e", 'Pkg.add("JLD"); using JLD')
+            if self.spec.version >= Version('1.0.0'):
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/julia/package.py]: @baberlevi it appears that all `julia` invocations after this point differ only in terms of the `using Pkg;` prefix. The logic could potentially be consolidated by prepending this for `julia` versions >= `1.0.0` (potentially with a function that constructs the `Pkg.add` command from the package name and `julia` version).

I'm merging this anyhow but thought I'd add a note for future edits to `julia`.

Comment (baberlevi) [reply var/spack/repos/builtin/packages/julia/package.py]: Thanks for the feedback, that's a good idea. I'll work on that and ping you to re-review

**Comment by alalazo** [comment]
@baberlevi Is this still relevant? In Spack v 0.13 I see we have already v 1.1.1:
```console
$ spack info julia
Package:   julia

Description:
    The Julia Language: A fresh approach to technical computing

Homepage: http://julialang.org

Tags: 
    None

Preferred version:  
    1.1.1          https://github.com/JuliaLang/julia/releases/download/v1.1.1/julia-1.1.1-full.tar.gz

[ ... ]
```

**Comment by alalazo** [comment]
Closing as stale. Feel free to reopen if you want to push this forward.

**Pull request closed without merging**